internal/middleware/compression.go

```diff
@@ -15,12 +15,69 @@
 package middleware
 
 import (
+	"bytes"
 	"compress/gzip"
 	"compress/zlib"
 	"io"
 	"net/http"
 )
 
+const (
+	headerContentEncoding = "Content-Encoding"
+	headerValueGZIP       = "gzip"
+)
+
+type CompressRoundTripper struct {
+	http.RoundTripper
+}
+
+func NewCompressRoundTripper(rt http.RoundTripper) *CompressRoundTripper {
```

Comment:
I will give a try after this PR merges to just do this directly in the exporter and see if we save some performance or code gets smaller.

Reply:
SGTM.

Reply:
It would be interesting to profile and see how much of a difference it makes (assuming we don't resintantiate gzipwriter on every request). It may be significant for very small requests (e.g. one span requests).